From e9159fd582157fb7083f66cd72a27e3701c930a8 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Thu, 30 Jul 2020 15:03:11 +0530
Subject: [PATCH] Bring back 'passwd' field in create_user

Many appliance teams are setting password using this method and looks
like mainline cloud-init doesn't support this anymore.

We need to revisit this patch once canonical team responds on 'passwd'
field

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 cloudinit/distros/__init__.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cloudinit/distros/__init__.py b/cloudinit/distros/__init__.py
index 2b559fe6..80c21905 100755
--- a/cloudinit/distros/__init__.py
+++ b/cloudinit/distros/__init__.py
@@ -531,6 +531,11 @@ class Distro(object):
         if 'hashed_passwd' in kwargs and kwargs['hashed_passwd']:
             self.set_passwd(name, kwargs['hashed_passwd'], hashed=True)
 
+        # TODO: This needs to be revisited after cloud-init team's response
+        # on 'passwd' field
+        if 'passwd' in kwargs and kwargs['passwd']:
+            self.set_passwd(name,kwargs['passwd'], True)
+
         # Default locking down the account.  'lock_passwd' defaults to True.
         # lock account unless lock_password is False.
         if kwargs.get('lock_passwd', True):
-- 
2.28.0

